pkg/kubelet/network/dns/dns.go

```diff
@@ -99,19 +102,31 @@ func omitDuplicates(strs []string) []string {
 func (c *Configurer) formDNSSearchFitsLimits(composedSearch []string, pod *v1.Pod) []string {
 	limitsExceeded := false
 
-	if len(composedSearch) > validation.MaxDNSSearchPaths {
-		composedSearch = composedSearch[:validation.MaxDNSSearchPaths]
+	for i, search := range composedSearch {
+		if len(search) > utilvalidation.DNS1123SubdomainMaxLength {
+			composedSearch = composedSearch[:i]
+			limitsExceeded = true
+			break
+		}
+	}
+
+	maxDNSSearchPaths, maxDNSSearchListChars := validation.MaxDNSSearchPathsLegacy, validation.MaxDNSSearchListCharsLegacy
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandedDNSConfig) {
+		maxDNSSearchPaths, maxDNSSearchListChars = validation.MaxDNSSearchPathsExpanded, validation.MaxDNSSearchListCharsExpanded
+	}
+	if len(composedSearch) > maxDNSSearchPaths {
```

Comment:
I understand Tim's point and I think he is right, if we truncate here the for in https://github.com/kubernetes/kubernetes/pull/100651/files#r607368307 doesn't have to go through searches that may be truncated later

Reply:
I've got your point. Done.